### PR TITLE
Wizard recipe: shaderc-v2022.1.0

### PR DESCRIPTION
**Pull request by jhigginbotham64**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: shaderc
* Version: v2022.1.0

@staticfloat please review and merge.

### CODE REVIEW

File: S/shaderc/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+./utils/git-sync-deps 
+
+mkdir build && cd build
+
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DSHADERC_SKIP_TESTS=1 -DSHADERC_SKIP_EXAMPLES=1 -DSHADERC_SKIP_COPYRIGHT_CHECK=1 -S .. -B .
+
+make -j${nproc}
+
+make install
```
Comment (giordano) [inline S/shaderc/build_tarballs.jl:25]:
```suggestion
./utils/git-sync-deps 
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DSHADERC_SKIP_TESTS=1 \
    -DSHADERC_SKIP_EXAMPLES=1 \
    -DSHADERC_SKIP_COPYRIGHT_CHECK=1 \
    -S .. \
    -B .
make -j${nproc}
make install
```

File: S/shaderc/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "shaderc"
```
Comment (giordano) [inline S/shaderc/build_tarballs.jl:5]:
```suggestion
name = "Shaderc"
```

File: S/shaderc/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    ExecutableProduct("spirv-reduce", :spirv_reduce),
+    ExecutableProduct("spirv-as", :spirv_as)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"11.1.0")
```
Comment (giordano) [inline S/shaderc/build_tarballs.jl:54]: Please don't use the latest version of GCC available, that can be a big compatibility issue.  Instead use the _oldest_ which can actually build the package.

File: S/shaderc/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DSHADERC_SKIP_TESTS=1 -DSHADERC_SKIP_EXAMPLES=1 -DSHADERC_SKIP_COPYRIGHT_CHECK=1 -S .. -B .
+
+make -j${nproc}
+
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline S/shaderc/build_tarballs.jl:30]: From the log:
```
┌ Warning: /agent/_work/1/s/S/shaderc/build/i686-linux-gnu/Mjo17nMt/i686-linux-gnu-libgfortran5-cxx11/destdir/bin/glslangValidator contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/Ke46L/src/auditor/compiler_abi.jl:247
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```
```

File: S/shaderc/build_tarballs.jl (hunk 3 above)
Comment (jhigginbotham64) [reply S/shaderc/build_tarballs.jl:56]: ok, thanks for pointing that out.

File: S/shaderc/build_tarballs.jl (hunk 4 above)
Comment (jhigginbotham64) [reply S/shaderc/build_tarballs.jl:32]: ok

**Comment by jhigginbotham64** [comment]
@giordano the name change appears to have broken something, but i'll let you troubleshoot and decide what to do. thanks again for all the help. :)

**Comment by giordano** [comment]
https://github.com/JuliaPackaging/Yggdrasil/pull/4611#discussion_r826423995

> Also, please note that you'll have to rename the directory accordingly

**Comment by jhigginbotham64** [comment]
whoops, quite right

**Pull request merged**